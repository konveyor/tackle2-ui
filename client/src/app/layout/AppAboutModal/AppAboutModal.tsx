import React from "react";
import { useTranslation, Trans } from "react-i18next";

import {
  AboutModal,
  TextContent,
  Text,
  TextVariants,
  TextList,
  TextListItem,
} from "@patternfly/react-core";

import tackleBrandImage from "@app/images/tackle.png";
import mtaBrandImage from "@app/images/logoRedHat.svg";
import { APP_BRAND, BrandType } from "@app/Constants";
import { ENV } from "@app/env";

export interface AppAboutModalProps {
  isOpen: boolean;
  onClose: () => void;
}

export const AppAboutModal: React.FC<AppAboutModalProps> = ({
  isOpen,
  onClose,
}) => {
  const { t } = useTranslation();
  const brandName =
    APP_BRAND === BrandType.Konveyor
      ? "Tackle"
      : "Migration Toolkit for Applications";
  return (
    <AboutModal
      isOpen={isOpen}
      onClose={onClose}
      trademark="COPYRIGHT Â© 2022."
      brandImageSrc={
        APP_BRAND === BrandType.Konveyor ? tackleBrandImage : mtaBrandImage
      }
      brandImageAlt="Logo"
      productName={brandName}
    >
      <TextContent>
        <Text component={TextVariants.h4}>{t("about.about")}</Text>
        <Text component={TextVariants.p}>
          {t("about.introduction", { brandType: brandName })}
        </Text>
        <Text component={TextVariants.p}>
          {t("about.description", { brandType: brandName })}
        </Text>
        <Text component={TextVariants.p}>
          {t("about.bottom1", { brandType: brandName })}{" "}
          <Text
            component={TextVariants.a}
            href="https://www.konveyor.io/"
            target="_blank"
          >
            Konveyor community
          </Text>
          .
        </Text>
        <Text component={TextVariants.p}>
          {t("about.bottom2")}{" "}
          <Text
            component={TextVariants.a}
            href={
              APP_BRAND === BrandType.Konveyor
                ? "https://tackle-docs.konveyor.io/"
                : "https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications"
            }
            target="_blank"
          >
            {brandName} documentation
          </Text>
          .
        </Text>
        <Text component={TextVariants.p}>
          <Trans i18nKey="about.iconLibrary">
            The Icon Library used in this project is a derivative of the{" "}
            <Text
              component={TextVariants.a}
              href="https://www.redhat.com/en/about/brand/standards/icons/standard-icons"
              target="_blank"
            >
              Standard Icons library
            </Text>
            by{" "}
            <Text
              component={TextVariants.a}
              href="https://www.redhat.com"
              target="_blank"
            >
              Red Hat
            </Text>
            , used under{" "}
            <Text
              component={TextVariants.a}
              href="https://creativecommons.org/licenses/by/4.0/"
              target="_blank"
            >
              CC BY 4.0
            </Text>
          </Trans>
        </Text>
      </TextContent>
      <TextContent className="pf-u-py-xl">
        <TextContent>
          <TextList component="dl">
            <TextListItem component="dt">{t("terms.version")}</TextListItem>
            <TextListItem component="dd">{ENV.VERSION}</TextListItem>
          </TextList>
        </TextContent>
      </TextContent>
    </AboutModal>
  );
};
