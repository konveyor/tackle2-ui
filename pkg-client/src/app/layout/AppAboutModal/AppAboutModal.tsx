import React from "react";
import { useTranslation, Trans } from "react-i18next";

import {
  AboutModal,
  TextContent,
  Text,
  TextVariants,
  TextList,
  TextListItem,
} from "@patternfly/react-core";

import brandImage from "@app/images/tackle.png";

const VERSION = "2.1";

export interface AppAboutModalProps {
  isOpen: boolean;
  onClose: () => void;
}

export const AppAboutModal: React.FC<AppAboutModalProps> = ({
  isOpen,
  onClose,
}) => {
  const { t } = useTranslation();

  return (
    <AboutModal
      isOpen={isOpen}
      onClose={onClose}
      trademark="COPYRIGHT Â© 2022."
      brandImageSrc={brandImage}
      brandImageAlt="Logo"
      productName="Tackle"
    >
      <TextContent className="pf-u-py-xl">
        <Text component={TextVariants.p}>{t("about.introduction")}</Text>
        <Text component={TextVariants.p}>{t("about.description")}</Text>
        <Text component={TextVariants.p}>
          <Trans i18nKey="about.bottom1">
            Tackle is a project within the{" "}
            <Text
              component={TextVariants.a}
              href="https://www.konveyor.io/"
              target="_blank"
            >
              Konveyor community
            </Text>
            .
          </Trans>
        </Text>
        <Text component={TextVariants.p}>
          <Trans i18nKey="about.bottom2">
            For more information please refer to{" "}
            <Text
              component={TextVariants.a}
              href="https://tackle-docs.konveyor.io/"
              target="_blank"
            >
              Tackle documentation
            </Text>
            .
          </Trans>
        </Text>
        <Text component={TextVariants.p}>
          <Trans i18nKey="about.iconLibrary">
            The Icon Library used in this project is a derivative of the{" "}
            <Text
              component={TextVariants.a}
              href="https://www.redhat.com/en/about/brand/standards/icons/standard-icons"
              target="_blank"
            >
              Standard Icons library
            </Text>
            by{" "}
            <Text
              component={TextVariants.a}
              href="https://www.redhat.com"
              target="_blank"
            >
              Red Hat
            </Text>
            , used under{" "}
            <Text
              component={TextVariants.a}
              href="https://creativecommons.org/licenses/by/4.0/"
              target="_blank"
            >
              CC BY 4.0
            </Text>
          </Trans>
        </Text>
      </TextContent>
      <TextContent className="pf-u-py-xl">
        <TextContent>
          <TextList component="dl">
            <TextListItem component="dt">{t("terms.version")}</TextListItem>
            <TextListItem component="dd">{VERSION}</TextListItem>
          </TextList>
        </TextContent>
      </TextContent>
    </AboutModal>
  );
};
