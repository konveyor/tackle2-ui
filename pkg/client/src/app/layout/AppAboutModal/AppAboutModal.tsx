import React from "react";
import { useTranslation, Trans } from "react-i18next";

import {
  AboutModal,
  TextContent,
  Text,
  TextVariants,
  TextList,
  TextListItem,
} from "@patternfly/react-core";

import brandImage from "@app/images/tackle.png";

export interface AppAboutModalProps {
  isOpen: boolean;
  onClose: () => void;
}

export const AppAboutModal: React.FC<AppAboutModalProps> = ({
  isOpen,
  onClose,
}) => {
  const { t } = useTranslation();

  return (
    <AboutModal
      isOpen={isOpen}
      onClose={onClose}
      trademark="COPYRIGHT Â© 2021."
      brandImageSrc={brandImage}
      brandImageAlt="Logo"
      productName="Tackle"
    >
      <TextContent className="pf-u-py-xl">
        <Text component={TextVariants.p}>{t("about.introduction")}</Text>
        <Text component={TextVariants.p}>{t("about.description")}</Text>
        <Text component={TextVariants.p}>
          <Trans i18nKey="about.bottom1">
            Tackle is a project within the{" "}
            <Text
              component={TextVariants.a}
              href="https://www.konveyor.io/"
              target="_blank"
            >
              Konveyor community
            </Text>
            .
          </Trans>
        </Text>
        <Text component={TextVariants.p}>
          <Trans i18nKey="about.bottom2">
            For more information please refer to{" "}
            <Text
              component={TextVariants.a}
              href="https://tackle-docs.konveyor.io/"
              target="_blank"
            >
              Tackle documentation
            </Text>
            .
          </Trans>
        </Text>
      </TextContent>
      <TextContent className="pf-u-py-xl">
        <TextContent>
          <TextList component="dl">
            <TextListItem component="dt">{t("terms.version")}</TextListItem>
            <TextListItem component="dd">1.2.0</TextListItem>
          </TextList>
        </TextContent>
      </TextContent>
    </AboutModal>
  );
};
