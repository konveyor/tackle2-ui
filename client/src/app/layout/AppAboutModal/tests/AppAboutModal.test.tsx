import { render } from "@testing-library/react";
import React from "react";
import { AppAboutModal } from "../AppAboutModal";

it("AppAboutModal", () => {
  const wrapper = render(<AppAboutModal isOpen={true} onClose={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});
