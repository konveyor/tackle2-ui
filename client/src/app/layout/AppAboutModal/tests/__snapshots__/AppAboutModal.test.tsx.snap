// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAboutModal 1`] = `
<AboutModal
  appendTo={null}
  backgroundImageSrc=""
  brandImageAlt="Logo"
  brandImageSrc="test-file-stub"
  className=""
  isOpen={true}
  noAboutModalBoxContentContainer={false}
  onClose={[MockFunction]}
  productName="Tackle"
  trademark="COPYRIGHT Â© 2022."
>
  <TextContent
    className="pf-u-py-xl"
  >
    <Text
      component="p"
    >
      about.introduction
    </Text>
    <Text
      component="p"
    >
      about.description
    </Text>
    <Text
      component="p"
    >
      about.bottom1
       
      <Text
        component="a"
        href="https://www.konveyor.io/"
        target="_blank"
      >
        Konveyor community
      </Text>
      .
    </Text>
    <Text
      component="p"
    >
      about.bottom2
       
      <Text
        component="a"
        href="https://tackle-docs.konveyor.io/"
        target="_blank"
      >
        Tackle
         documentation
      </Text>
      .
    </Text>
    <Text
      component="p"
    >
      <Component
        i18nKey="about.iconLibrary"
      >
        The Icon Library used in this project is a derivative of the
         
        <Text
          component="a"
          href="https://www.redhat.com/en/about/brand/standards/icons/standard-icons"
          target="_blank"
        >
          Standard Icons library
        </Text>
        by
         
        <Text
          component="a"
          href="https://www.redhat.com"
          target="_blank"
        >
          Red Hat
        </Text>
        , used under
         
        <Text
          component="a"
          href="https://creativecommons.org/licenses/by/4.0/"
          target="_blank"
        >
          CC BY 4.0
        </Text>
      </Component>
    </Text>
  </TextContent>
  <TextContent
    className="pf-u-py-xl"
  >
    <TextContent>
      <TextList
        component="dl"
      >
        <TextListItem
          component="dt"
        >
          terms.version
        </TextListItem>
        <TextListItem
          component="dd"
        >
          99.0.0
        </TextListItem>
      </TextList>
    </TextContent>
  </TextContent>
</AboutModal>
`;
